### DISCUSSION TITLE
Labo 2 - RentalTestHelper Error

### DISCUSSION THREAD

**Discussion by JoelMatias** [amt-classroom/amt-classroom.github.io#14] [General]
Bonjour, mon programme crash lors de l'appel de la fonction suivante se trouvant dans RentalTestHelper : 
![image](https://github.com/amt-classroom/amt-classroom.github.io/assets/73992136/ead4ab51-b5fc-48b0-b817-9317cb2c9494)

Tout les autres tests de RentalTestService fonctionnent du coup à pars testRental()

Voici l'erreur :
![image](https://github.com/amt-classroom/amt-classroom.github.io/assets/73992136/b4e99bfa-9a2b-47d1-bdde-d19e7c791b92)

**Comment by Perdjesk** [comment]
Merci de mettre la version textuelle de la trace d'erreur au complet.

De plus donner une référence stable (commit) dans votre dépôt depuis laquelle le cas serait reproduisible.

**Reply by JoelMatias** [reply]
C'est tout bon, le problème était lié à ma version de java SDK (21). Mais les tests fonctionnent. Je supprimerais cette discussion étant inutile

**Reply by Perdjesk** [reply]
Je ne pense pas que votre erreur soit liée à la version du SDK Java.

Par contre vos changements sur l'entité `Rental` sont sûrement la résolution à votre problème.  https://github.com/amt-classroom/lab-2-business-layer-marzullo_matias_escher/commit/831804f6c03729ca72827e49915883bcb0263108

```
-   private Timestamp return_date;
+   private Timestamp returnDate;
```

La version pour ces requêtes utilise JPQL sans sécurité à la compilation. Comme votre code utilisait des noms de champs différent le parsing de la requête échouait. 
Je prends note de transformer ces requêtes en version SQL afin d'éviter de dépendre des entités JPA pour la future version du laboratoire.
